src/libraries/System.Text.Json/tests/System.Text.Json.Tests/JsonDocumentTests.cs

```diff
@@ -3458,11 +3458,34 @@ public static void ParseValue_AllowMultipleValues_TrailingContent()
         [InlineData("""{ "foo" : [1,2,3,4] }""", 1)]
         [InlineData("""{}""", 0)]
         [InlineData("""{ "foo" : {"nested:" : {"nested": 1, "bla": [1, 2, {"bla": 3}] } }, "test": true, "foo2" : {"nested:" : {"nested": 1, "bla": [1, 2, {"bla": 3}] } }}""", 3)]
-        public static void TestGetPropertyCount(string json, int expectedCount) {
+        public static void TestGetPropertyCount(string json, int expectedCount)
+        {
             JsonElement element = JsonSerializer.Deserialize<JsonElement>(json);
             Assert.Equal(expectedCount, element.GetPropertyCount());
         }
 
+        [Fact]
+        public static void VerifyGetPropertyCountUsingEnumerateObject()
+        {
+            using (JsonDocument doc = JsonDocument.Parse(SR.ProjectLockJson))
+            {
+                CheckPropertyCountAgainstEnumerateObject(doc.RootElement);
+            }
+
+            void CheckPropertyCountAgainstEnumerateObject(JsonElement obj)
+            {
+                Assert.Equal(obj.EnumerateObject().Count(), obj.GetPropertyCount());
+
+                foreach (JsonProperty prop in obj.EnumerateObject())
+                {
+                    if (prop.Value.ValueKind == JsonValueKind.Object)
+                    {
+                        CheckPropertyCountAgainstEnumerateObject(prop.Value);
+                    }
```

Comment:
Should there also be something like

```
if (prop.Value.ValueKind == JsonValueKind.Array) {
foreach item, check
}
```
?

Reply:
How did I miss that... I now also added checks for `GetArrayLength` (because the local variable is now reused/shared for object property count and array length here: https://github.com/dotnet/runtime/commit/ea276ea6ba43a52c8380bf8d6c074d815a12d289)